app/javascript/controllers/webpush_subscription_controller.js

```diff
@@ -0,0 +1,38 @@
+import { Controller } from "stimulus"
+
+export default class extends Controller {
+  static values = {
+    url: String,
+    vapidPublicKey: Array
+  }
+
+  connect() {
+    const vapidPublicKey = new Uint8Array(this.vapidPublicKeyValue);
+    const url = this.urlValue
+    if (navigator.serviceWorker) {
+      navigator.serviceWorker.register('/service_worker.js')
+      .then(function(reg) {
+        navigator.serviceWorker.ready.then((serviceWorkerRegistration) => {
+          serviceWorkerRegistration.pushManager
+          .subscribe({
+            userVisibleOnly: true,
+            applicationServerKey: vapidPublicKey
+          }).then(async function(sub){
+            const data = await fetch(url, {
+              method: 'POST',
+              headers: {
+                'Content-Type': 'application/json'
+              },
+              body: JSON.stringify(sub)
+            })
```

Comment:
Acho que aqui pode ser interessante utilizarmos essa lib
https://github.com/rails/request.js#how-to-use
https://gorails.com/episodes/rails-requestjs

Ela já adiciona os detalhes de autenticação na request

Reply:
Interessante!